frame/bounties/src/lib.rs

```diff
@@ -0,0 +1,762 @@
+						ensure!(*curator == signer, Error::<T>::RequireCurator);
+						*update_due = (system::Module::<T>::block_number() + T::BountyUpdatePeriod::get()).max(*update_due);
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				}
+
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::<T>::BountyExtended(bounty_id));
+		}
+	}
+}
+
+impl<T: Trait> Module<T> {
+	// Add public immutables and private mutables.
+
+	/// The account ID of the treasury pot.
+	///
+	/// This actually does computation. If you need to keep using it, then make sure you cache the
+	/// value and only call this once.
+	pub fn account_id() -> T::AccountId {
+		T::ModuleId::get().into_account()
+	}
+
+	/// The account ID of a bounty account
+	pub fn bounty_account_id(id: BountyIndex) -> T::AccountId {
+		// only use two byte prefix to support 16 byte account id (used by test)
+		// "modl" ++ "py/trsry" ++ "bt" is 14 bytes, and two bytes remaining for bounty index
+		T::ModuleId::get().into_sub_account(("bt", id))
+	}
+
+	fn create_bounty(
+		proposer: T::AccountId,
+		description: Vec<u8>,
+		value: BalanceOf<T>,
+	) -> DispatchResult {
+		ensure!(description.len() <= T::MaximumReasonLength::get() as usize, Error::<T>::ReasonTooBig);
+		ensure!(value >= T::BountyValueMinimum::get(), Error::<T>::InvalidValue);
+
+		let index = Self::bounty_count();
+
+		// reserve deposit for new bounty
+		let bond = T::BountyDepositBase::get()
+			+ T::DataDepositPerByte::get() * (description.len() as u32).into();
+		T::Currency::reserve(&proposer, bond)
+			.map_err(|_| Error::<T>::InsufficientProposersBalance)?;
+
+		BountyCount::put(index + 1);
+
+		let bounty = Bounty {
+			proposer,
+			value,
+			fee: 0u32.into(),
+			curator_deposit: 0u32.into(),
+			bond,
+			status: BountyStatus::Proposed,
+		};
+
+		Bounties::<T>::insert(index, &bounty);
+		BountyDescriptions::insert(index, description);
+
+		Self::deposit_event(RawEvent::BountyProposed(index));
+
+		Ok(())
+	}
+
+}
+
+impl<T: Trait> pallet_treasury::SpendFunds<T> for Module<T> {
+	fn spend_funds(
+		budget_remaining: &mut BalanceOf<T>,
+		imbalance: &mut PositiveImbalanceOf<T>,
+		total_weight: &mut Weight,
+		missed_any: &mut bool
+	) {
+		let bounties_len = BountyApprovals::mutate(|v| {
+			let bounties_approval_len = v.len() as u32;
+			v.retain(|&index| {
+				Bounties::<T>::mutate(index, |bounty| {
+					// Should always be true, but shouldn't panic if false or we're screwed.
+					if let Some(bounty) = bounty {
+						if bounty.value <= *budget_remaining {
+							*budget_remaining -= bounty.value;
+
+							bounty.status = BountyStatus::Funded;
+
+							// return their deposit.
+							let _ = T::Currency::unreserve(&bounty.proposer, bounty.bond);
+
+							// fund the bounty account
+							imbalance.subsume(T::Currency::deposit_creating(&Self::bounty_account_id(index), bounty.value));
+
+							Self::deposit_event(RawEvent::BountyBecameActive(index));
+							false
```

Comment:
Might be a logic error here. "missed_any" may come into this loop as true but here it would turn to false, even though it should keep its state as true.

Reply:
For me logic occurs OK, Shawn, state of "missed_any" gets updated only once at line 748. it's not overwritten by false.

![](https://i.imgur.com/0LYpmNM.png)

Plz correct me if I am wrong ...

Reply:
What I am saying is that `missed_any` can come from `pallet_treasury` having been true/false, and then changed here? Ill need to look again if this does not make sense